feat: Repeating ToDo uses Auto Repeat

iRaySpace opened the pull request:
**Triggers repeating task**
![image](https://user-images.githubusercontent.com/5818544/56482704-16e0a980-64f8-11e9-9a3b-c7da50e45bf3.png)

**Create Auto Repeat on ToDo**
![image](https://user-images.githubusercontent.com/5818544/56482695-04ff0680-64f8-11e9-8c21-6eee509ef23c.png)

karthikeyan5 committed:
feat(Raw Printing): Adding Support for qz-tray

karthikeyan5 committed:
fix(Raw Printing): fixing code standard

karthikeyan5 committed:
fix(raw printing): codacy issues

karthikeyan5 committed:
fix(raw printing): more codacy fixes

barredterra committed:
feat(integrations): add autocomplete to gsuite templates

scmmishra committed:
feat: Added template for imageview_fields

karthikeyan5 committed:
fix(Raw Printing): fixes as per PR review

> fixing confusing naming in the UI and calling it "Raw Print"
> used import for sha256 library
> changed the Print Preview message for raw print
> same possible codacy fixes

karthikeyan5 committed:
fix(raw printing): more codacy fixes

karthikeyan5 committed:
feat(Raw Printing): added "Enable Raw Printing"

> added more comments in code

scmmishra committed:
fix: inconsistent padding in employee image view

scmmishra committed:
fix: Template for image view details

netchampfaris committed:
feat: Introduce new upload dialog

netchampfaris committed:
feat: Use Feature Flag to enable new upload

netchampfaris committed:
fix: Missing semicolon

netchampfaris committed:
fix: Add file type and size restrictions

netchampfaris committed:
fix: Upload via web link

barredterra committed:
put functions in scope

netchampfaris committed:
fix: Pass file_doc to on_success

netchampfaris committed:
refactor: Rename multiple_files to allow_multiple

netchampfaris committed:
fix: Replace old upload with new FileUploader

netchampfaris committed:
fix: Refactor Attach controls to use new FileUploader

netchampfaris committed:
fix: Remove Feature Flag settings

netchampfaris committed:
test: Add UI test for FileUploader

netchampfaris committed:
style: Missing semicolon

netchampfaris committed:
fix: Force click web link

barredterra committed:
function out of scope, but with unique name

barredterra committed:
make ID fields read only

barredterra committed:
rm unnecessary functions

netchampfaris committed:
fix: Cleanup modal after hide

netchampfaris committed:
test: Wait for modal to hide

scmmishra committed:
feat: Refactored image view details

- Now "Show in List View" marked fields are shown in the image view

scmmishra committed:
fix: Bottom border not visible for single item in image view

surajshetty3416 committed:
fix: save button should not remain disabled when the system is offline

scmmishra committed:
fix (codacy): Fixed linting issues

karthikeyan5 committed:
feat(Raw Printing): added help text in UI

Mangesh-Khairnar committed:
feat: add a is default field in user permission

Mangesh-Khairnar committed:
style: change comment formatting

Mangesh-Khairnar committed:
feat: add validation for overlap default entries

Mangesh-Khairnar committed:
feat: assign default value to link field based on the user permission

Mangesh-Khairnar committed:
feat: fetch default value for link field based on user permission

rmehta committed:
fix(comment): blog comment email and min length

deepeshgarg007 committed:
fix: Do not rename or move files for custom doctype

surajshetty3416 committed:
fix: Ignore permission while saving revert doc

- Since on system manager can revert points

rmehta committed:
feat(get_period_beginning): in frappe.desk.doctype.dashboard_chart.dashboard_chart

surajshetty3416 committed:
fix: Disable submit button while doing network request
- to avoid duplicate entries

deepeshgarg007 committed:
fix: Custom columns handling for prepared report

netchampfaris committed:
fix: Show help text for public/private toggle

netchampfaris committed:
fix: Alignment

netchampfaris committed:
fix: Show Hidden fields in Report Builder

netchampfaris committed:
fix: Allow numeric filter without using =

karthikeyan5 committed:
fix(Raw Printing): fixes as per PR review 2

> qz-tray.js and sha256.min.js are loaded only on demand
> renamed 'raw print settings' button to 'Printer Settings'
> removed commented code
> fixed spelling mistakes

rmehta committed:
fix(test): comments

rmehta committed:
fix: link selection relevance

netchampfaris committed:
feat: Full Width Container

Persistent using localStorage since container width should be per device

netchampfaris committed:
fix: Remove projectId to disable recording

surajshetty3416 committed:
fix: Change fieldtype of assign_condition from "Small Text" to "Code"

- Small text fields gets sanitized unless ignore_xss_filter
 is explicitly enabled.
- Code fieldtype seems to be right fieldtype for such use case

netchampfaris committed:
fix: Remove --record flag

Mangesh-Khairnar committed:
feat: pass default value to create user permission via user permission list

Mangesh-Khairnar committed:
style: change formatting for is default

Mangesh-Khairnar committed:
feat: add is default option in add/update dialog in user permission list

surajshetty3416 committed:
fix: Remove default for user from review dialog

- let users explicitly select whom they want to review

Mangesh-Khairnar committed:
feat: pass default value to add user permission

Mangesh-Khairnar committed:
test: check default user permission overlap validation

Mangesh-Khairnar committed:
test: validate default value fetch in multiple user permission case

Mangesh-Khairnar committed:
fix: change raw sql to orm

ercyams committed:
feat: auto repeat todo

ercyams committed:
feat: auto repeat todo

rmehta committed:
feat: Milestone Tracker. Track document lifecycle with milestones

Commit:
fix(mariadb): refactor db setup (#7049)

* Fixed setup_db.py to support MariaDB version 10.3.  Added function for calculating versions, per MariaDB documentation.

* Major rewrite of setup_db.py. Reduced number of functions. Made fully compatible with MariaDB 10.3+. Removed references to client parameters in my.cnf that cannot be queried.  Applied PEP8 rules to python code.

* Adding missing print statement in setup_db.py

* Replaced 4-spaces with tabs, per ERPNext standards.

* Fixed broken test by substituing 'check_database_settings()'

Fixed the test in convert_to_barracuda_and_utf8mb4.py by calling the new function 'check_database_settings()'

* Cleaned up setup.py, to match latest version of Frappe develop

SaiFi0102 committed:
feat: Use `field_order` attr in DocType JSON file for cleaner git diffs

SaiFi0102 committed:
test(DocType): test_load_file_field_order

Mangesh-Khairnar committed:
fix: default user permission to handle small int values

rmehta committed:
fix: API change for version

rmehta committed:
fix(tests)

rohitwaghchaure committed:
fix: On assignment, system sending an email to the assigned user even if notify to email is disabled

rohitwaghchaure committed:
fix: User not able to make lead from the communication

SaiFi0102 committed:
fix: Move field_order sync logic to DocType controller and introduce hooks
before_export, before_import and prepare_docdict_for_import

SaiFi0102 committed:
feat: Remove null and empty attributes from DocType JSON file on export

SaiFi0102 committed:
fix: test

SaiFi0102 committed:
fix: Remove trailing spaces from exported JSON

SaiFi0102 committed:
fix: test

netchampfaris committed:
feat: Dynamic Date filters in Auto Email Report

SaiFi0102 committed:
fix: test

SaiFi0102 committed:
fix: codacy

netchampfaris committed:
fix: Test

surajshetty3416 committed:
fix: Do not set page_length as null (#7314)

Do not set page_length as null since due to this all the records are fetched at once which slows down the system if there are thousands of records

netchampfaris committed:
test: Dynamic Dates in Auto Email Report

netchampfaris committed:
style: Method could be a function

netchampfaris committed:
fix: Remove unused import

netchampfaris committed:
refactor: Remove frappe.upload.make

netchampfaris committed:
fix: Re-enable cypress test recording (#7319)

netchampfaris committed:
fix: Print Format UX (#7284)

- Rename type Server and Client to Jinja
- Rename Js to JS
- Patch
- Reload Print Format when new one is created
- Missing Semicolon and sanity check

netchampfaris committed:
fix: Revert backslash in templates

surajshetty3416 committed:
test: fix fill_field command for code fieldtype

rmehta committed:
fix(for-patch): cache manager to ignore missing tables

rmehta committed:
feat(grid): Move row to a particular id

deepeshgarg007 committed:
fix: Link field fixes for custom reports

Mangesh-Khairnar committed:
fix: return allowed docs along with default doc

Commit:
Allow user to specify database port

rmehta committed:
fix(linting)

iRaySpace committed:
refactor: added new field for ui and is_repeating flow

iRaySpace committed:
refactor: added new field for ui and is_repeating flow

iRaySpace committed:
refactor: fix indentation for codacy

iRaySpace committed:
refactor: fix indentation for codacy

iRaySpace committed:
refactor: 2nd fix for indent

iRaySpace committed:
refactor: 2nd fix for indent

iRaySpace committed:
refactor: fields to const `fields`, converted spaces to tabs

iRaySpace committed:
refactor: fields to const `fields`, converted spaces to tabs

Anurag810 committed:
fix: Cannot delete Issue because it is linked to Comments(v12) (#7270)

* fix: Cannot delete Issue because it is linked to Comments

* fix: commonified code

netchampfaris committed:
fix: Grid Upload

Commit:
Resolve conversation

adityahase committed:
feat: Maintain list of tables touched during migrate

This can be used to selectively restore changed tables from backup
after migrate failure

prasadarr committed:
fix broken YouTube link for user help

Mangesh-Khairnar committed:
fix: filter default docs from allowed docs

rohitwaghchaure committed:
fix: moved erpnext related methods from frappe to erpnext (#7332)

rmehta committed:
fix(milestone_tracker): doctype ui

Mangesh-Khairnar committed:
style: change permissions format

rohitwaghchaure committed:
fix: Prepared report shows status as Queued even if job is failed (#7331)

rmehta commented on frappe/desk/doctype/todo/todo.js:
```diff
@@ -1,40 +1,90 @@
 		}
+	},
+
+	new_auto_repeat_prompt: function (frm) {
```
we can add a generic `allow_auto_repeat` and `auto_repeat_date` property on DocType so that we can make any document ready for auto-repeat. This can be then extended to other objects like Task, etc.

The code will be very similar

iRaySpace replied:
Hello, I have confusion in `auto_repeat_date`. Is this a starting date or an end date?

iRaySpace replied:
Are the properties to be made only on `ToDo` or the general DocType?

rmehta replied:
general DocType

auto-repeat should be for start date (ideally!)

Move this to a common file

iRaySpace replied:
Thank you for pointing out! :grinning:

iRaySpace replied:
Hello,
I have made some changes below. I didn't add `auto_repeat_date`, I feel like it is redundant. Since Auto Repeat requires any DocType to have `auto_repeat` field.

rmehta commented on frappe/desk/doctype/todo/todo.js:
```diff
@@ -1,40 +1,90 @@
+			{
+				'fieldname': 'frequency',
+				'fieldtype': 'Select',
+				'options': ['Daily', 'Weekly', 'Monthly', 'Quarterly', 'Half-yearly', 'Yearly'],
```
translatable

```
'options': [{label:__('Daily'), value: 'daily'}, .....
```

rmehta commented on frappe/desk/doctype/todo/todo.js:
```diff
@@ -1,40 +1,90 @@
+				'fieldname': 'frequency',
+				'fieldtype': 'Select',
+				'options': ['Daily', 'Weekly', 'Monthly', 'Quarterly', 'Half-yearly', 'Yearly'],
+				'label': 'Frequency',
```
__('Frequency')

All user labels should be translatable

rmehta commented on frappe/desk/doctype/todo/todo.js:
```diff
@@ -1,40 +1,90 @@
+				},
+				callback: function (r) {
+					if (r.message) {
+						frappe.msgprint("Successfully created repeating task.", "Auto Repeat");
```
use `frappe.show_alert` and translatable

rmehta commented on frappe/desk/doctype/todo/todo.py:
```diff
@@ -108,4 +108,17 @@ def new_todo(description):
+
+
+@frappe.whitelist()
+def new_auto_repeat(todo, start_date, end_date, frequency):
```
make generic

iRaySpace replied:
Extended Auto Repeat's `make_auto_repeat` function

rmehta reviewed (changes requested):
👍 for first contribution, this can be made more generic!

surajshetty3416 committed:
fix: Remove extra call for review points of form load

surajshetty3416 committed:
fix: Remove review section from attachments

- Make separate section for review

surajshetty3416 committed:
fix: Replace single quotes with double

iRaySpace committed:
refactor: translatables and extended make_auto_repeat function

iRaySpace committed:
refactor: translatables and extended make_auto_repeat function

rmehta committed:
fix(style): cleaner style for form sidebar and other minor cleanups

rmehta committed:
fix: remove console.log

surajshetty3416 committed:
fix: Remove frappe.upload.get_strting

surajshetty3416 committed:
fix: Decode file data to string

Commit:
Resolve conversation

Commit:
Resolve conversation

Commit:
Resolve conversation

nabinhait committed:
fix: Removed get_doc_before_save

netchampfaris committed:
fix: Attach get_value

Return file_url in get_value

netchampfaris committed:
fix: Page Title wrapping and centering

scmmishra committed:
fix: Major bugs in query_report.js

scmmishra committed:
fix (linitng): codacy

surajshetty3416 committed:
feat: Show Dialog to edit profile data

surajshetty3416 committed:
fix: Realtime image update on image change from edit profile dialog

- Remove unwnated code

surajshetty3416 committed:
feat: Add period filter and user search option

surajshetty3416 committed:
style: Fix formatting

surajshetty3416 committed:
fix: Pass from date to energy_point_history

- So that it can only show logs of current point

scmmishra committed:
chore: Minor code improvements

surajshetty3416 committed:
fix: add a more descriptive method name

Co-Authored-By: Mangesh-Khairnar <mkhairnar10@gmail.com>

netchampfaris committed:
fix(ReportView): Remember column widths

Mangesh-Khairnar committed:
test: validate no fetch for multiple non default user permission

SaiFi0102 committed:
style: Rename prepare_docdict_for_import to prepare_for_import

Mangesh-Khairnar committed:
style: formatting changes

netchampfaris committed:
Update frappe/public/js/frappe/views/reports/query_report.js

Fixed syntax error

Co-Authored-By: scmmishra <scmmishra@users.noreply.github.com>

Mangesh-Khairnar committed:
style(linting): minor fixes

rmehta committed:
fix(style): tighter whitespace in sidebar

SaiFi0102 committed:
style: Linting for boilerplate files

scmmishra committed:
style: minor improvements

scmmishra committed:
fix: linting fixes

surajshetty3416 committed:
fix: Remove quotes around color hex

rmehta committed:
feat: allow addition of buttons in list

rmehta committed:
fix(minor): style and linting

rmehta committed:
fix(grid): drag to reorder was broken. fixed.

surajshetty3416 committed:
feat: Add a rank column for leaderboard (#7356)

scmmishra committed:
fix: fixed Make new button for empty state in report view

netchampfaris committed:
feat: Index all web pages for search (#7322)

* feat: Index Web Pages for search

- Walks and indexes all webpages in global_search

* test: Web Page Indexing

adityahase committed:
perf: Remove MANIFEST.in (#7360)

Faster pip install -e frappe

https://stackoverflow.com/questions/24727709/do-python-projects-need-a-manifest-in-and-what-should-be-in-it

netchampfaris committed:
fix: Method docstring

netchampfaris committed:
fix(rollup): Use terser instead of uglify

surajshetty3416 committed:
fix: Call week_start function to get week start date

iRaySpace committed:
merge from develop

nexusfidelity committed:
refactor: use utils new_auto_repeat_prompt

nexusfidelity committed:
refactor: use utils new_auto_repeat_prompt

nexusfidelity committed:
refactor: use utils new_auto_repeat_prompt

nexusfidelity committed:
refactor: use utils new_auto_repeat_prompt

nexusfidelity committed:
fix: code format

nexusfidelity committed:
fix: code format

nexusfidelity committed:
fix: code format of utils

nexusfidelity committed:
fix: code format of utils

nexusfidelity committed:
fix: code format of utils

nexusfidelity committed:
fix: code format of utils

iRaySpace committed:
refactor: moved new_auto_repeat_prompt to common.js, from utils.js

iRaySpace committed:
refactor: moved new_auto_repeat_prompt to common.js, from utils.js

iRaySpace committed:
feat: allow_auto_repeat on doctype; repeat on menu; removed todo repeat

iRaySpace committed:
feat: allow_auto_repeat on doctype; repeat on menu; removed todo repeat

iRaySpace commented:
**Added Repeat Option on Menu**
![image](https://user-images.githubusercontent.com/5818544/57205473-7d35f380-6ff1-11e9-9c53-7f5bc4316ddb.png)

**Added Allow Auto Repeat field on DocType**
![image](https://user-images.githubusercontent.com/5818544/57205498-a9517480-6ff1-11e9-9227-a52599695be8.png)

**Others Features**
- Added `auto_repeat` field on ToDo to make Auto Repeat

rmehta commented:
@iRaySpace hey, thanks, can you rebase this?

rmehta commented on frappe/desk/doctype/auto_repeat/auto_repeat.py:
```diff
@@ -354,13 +367,13 @@ def auto_repeat_doctype_query(doctype, txt, searchfield, start, page_len, filter
-		})
+		'key': searchfield,
+	}), {
+							 'txt': "%%%s%%" % txt,
```
please use only 1 indent, not multiple

iRaySpace replied:
Okay! I will fix the conflict.

rmehta replied:
@iRaySpace reminder :)

iRaySpace replied:
Hello @rmehta Working on rebase. Thank you for reminding.

iRaySpace replied:
Hello @rmehta I think I should re-do my customization by new branch based from the latest develop and open a new pull request. It seems my rebase is done wrong.

iRaySpace committed:
Merge branch 'auto_repeat_todo' of https://github.com/iRaySpace/frappe into auto_repeat_todo

netchampfaris commented:
Refactored in https://github.com/frappe/frappe/pull/7820
